### ISSUE TITLE
If no ZIM found for a language then automatically make the search with other languages (kiwix-serve)

### ISSUE THREAD

**Issue by kelson42** [kiwix/kiwix-tools#562]

**Comment by kelson42** [comment]
@juuz0 Do you have an opinion on this idea?

**Comment by juuz0** [comment]
@kelson42 I think it's good and definitely useful (as I had the exact same problem as yours earlier).

Though, for the implementation, not too sure if we should be showing tiles straight away (feels like that sort of defeats the purpose of filter). Alternatively, how about showing a link after all related tiles end:
"X results found for term 'xyz' in other languages, click here to display them." 
After that, display them (but I also feel these are extra clicks for the same thing as you suggested).

**Comment by kelson42** [comment]
I don't it would defeat the purpose of language filter if we would "always" provide as well results in other languages. It is just that the language filter won't be strict, but kind of "prefered result".

I'm in favor to run the search for all languages also only if all tiles for the prefered language are already visible ; either from the beginning or after scrolling down.

Last point: in last section, the tiles for all languages should not be displayed, only for all languages without the prefered language.

**Comment by juuz0** [comment]
Yeah okay I agree, should I start implementing it then? @kelson42

**Comment by kelson42** [comment]
@juuz0 Not sure. Would ne great, but don't know how complex it is.

**Comment by mgautierfr** [comment]
Could it be a sorting order (queried language first, other language after) instead of a true filter ?

**Comment by kelson42** [comment]
@mgautierfr I don’t think so. Ultimatively we will probably have a language filter accepting many languages.

**Comment by juuz0** [comment]
If this is still desirable, I can start working on this. 

Extra:
This line got my attention,
> Kiwix-serve welcome page search, per default, runs with the browser language. It’s not possible (for the moment), to make search in many chosen languages…. Or then with all of them.

Seems to be possible now (https://github.com/kiwix/libkiwix/pull/841) - no possible way to do it through UI yet (which shouldn't be much hard again, multiple checkboxes? :> don't see an issue open for that though)

**Comment by mgautierfr** [comment]
I think that search in multiple (selected) languages is a different issue.

Another solution to the current problem (the title of the issue is more a proposed solution to a problem than a description of the problem :) ) is that we could do the search in all language by default and propose a button `search in <your language> only`  to help user filtering more the results.

**Comment by kelson42** [comment]
> If this is still desirable, I can start working on this.

Yes. Let me rephrase this ticket:
* We need the features
* All the time (not only if no results in preselected language), but only if there is space to display these other books and if there is a language filter selected
* For the moment you probably need to make two requests (one with the language filter, one without)
* Visually we need a clear separation between the books with the language filter and the one without so first "For your language(s)" like today on the top and then "In other languages" below.
